### PR TITLE
test(fips): added a 4h 100gb longevity with fips enabled + artifact fixes

### PR DESCRIPTION
**Pull request by ShlomiBalalis**
FIPS are a set of standards for encryption algorithms (among others). To verify that Scylla complies with all fips completely, I created this scenario which enables every kind of encryption scylla supports, while FIPS is enabled

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: jenkins-pipelines/longevity-100gb-4h-fips.jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!groovy
+
+def lib = library identifier: 'sct@snapshot', retriever: legacySCM(scm)
+
+// This scenario can only be triggered with a scylla repo, not an ami
+longevityPipeline(
+    backend: 'aws',
+    aws_region: 'eu-west-1',
+    test_name: 'longevity_test.LongevityTest.test_custom_time',
+    test_config: '''["test-cases/longevity/longevity-100gb-4h.yaml", "configurations/longevity-fips-and-encryptions.yaml", "configurations/local-ear.yaml"]'''
```
Comment (fgelcer) [inline jenkins-pipelines/longevity-100gb-4h-fips.jenkinsfile:10]: first time i see 3 config files!!! hope it works flawlessly

**Review by fgelcer — APPROVED** [review]
LGTM

File: jenkins-pipelines/longevity-100gb-4h-fips.jenkinsfile (hunk 1 above)
Comment (ShlomiBalalis) [reply jenkins-pipelines/longevity-100gb-4h-fips.jenkinsfile:10]: Exactly what I thought! I am making history

**Comment by ShlomiBalalis** [comment]
Job currently failing:
```
f:log.py          l:23   c:sdcm.utils.log       p:ERROR > Traceback (most recent call last):
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/./sct.py", line 1741, in <module>
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     cli.main(prog_name="hydra")
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1055, in main
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     rv = self.invoke(ctx)
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     return _process_result(sub_ctx.command.invoke(sub_ctx))
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     return ctx.invoke(self.callback, **ctx.params)
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 760, in invoke
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     return __callback(*args, **kwargs)
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/./sct.py", line 1106, in run_test
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     unittest.main(module=None, argv=['python -m unittest', argv],
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/unittest/main.py", line 100, in __init__
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     self.parseArgs(argv)
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/unittest/main.py", line 147, in parseArgs
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     self.createTests()
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/unittest/main.py", line 158, in createTests
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     self.test = self.testLoader.loadTestsFromNames(self.testNames,
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/unittest/loader.py", line 220, in loadTestsFromNames
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     suites = [self.loadTestsFromName(name, module) for name in names]
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/unittest/loader.py", line 220, in <listcomp>
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     suites = [self.loadTestsFromName(name, module) for name in names]
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/unittest/loader.py", line 198, in loadTestsFromName
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     inst = parent(name)
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/sdcm/tester.py", line 292, in __init__
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     self._init_params()
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/sdcm/tester.py", line 665, in _init_params
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     self.params = init_and_verify_sct_config()
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/sdcm/sct_config.py", line 2331, in init_and_verify_sct_config
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     sct_config.verify_configuration()
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/sdcm/sct_config.py", line 2014, in verify_configuration
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     self._check_version_supplied(backend)
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/sdcm/sct_config.py", line 2148, in _check_version_supplied
f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     assert any(self.get(o) for o in one_of_options_must_exist), \
f:log.py          l:23   c:sdcm.utils.log       p:ERROR > AssertionError: scylla version/repos wasn't configured correctly
f:log.py          l:23   c:sdcm.utils.log       p:ERROR > configure one of the following: ['ami_id_db_scylla']
f:log.py          l:23   c:sdcm.utils.log       p:ERROR > or one of those environment variables: ['SCT_AMI_ID_DB_SCYLLA']
```

**Comment by ShlomiBalalis** [comment]
Job running: https://jenkins.scylladb.com/job/scylla-staging/job/Shlomo/job/longevity-100gb-4h-fips-test/4/

**Comment by ShlomiBalalis** [comment]
> Job running: https://jenkins.scylladb.com/job/scylla-staging/job/Shlomo/job/longevity-100gb-4h-fips-test/4/

Run currently DOA:
```
2023-08-06 16:29:37.214: (TestFrameworkEvent Severity.ERROR) period_type=one-time event_id=ca6bea68-6a62-4651-ba69-cba7beb49058, source=LongevityTest.test_custom_time (longevity_test.LongevityTest)() message=Traceback (most recent call last):
File "/home/ubuntu/scylla-cluster-tests/longevity_test.py", line 95, in test_custom_time
self.run_pre_create_schema()
File "/home/ubuntu/scylla-cluster-tests/longevity_test.py", line 71, in run_pre_create_schema
self._pre_create_schema(keyspace_num, scylla_encryption_options=self.params.get(
File "/home/ubuntu/scylla-cluster-tests/longevity_test.py", line 359, in _pre_create_schema
self.create_table(name='standard1', keyspace_name=keyspace_name, key_type='blob', read_repair=0.0, compact_storage=True,
File "/home/ubuntu/scylla-cluster-tests/sdcm/tester.py", line 2293, in create_table
session.execute(query)
File "/home/ubuntu/scylla-cluster-tests/sdcm/utils/common.py", line 1751, in execute_verbose
return execute_orig(*args, **kwargs)
File "cassandra/cluster.py", line 2699, in cassandra.cluster.Session.execute
File "cassandra/cluster.py", line 5018, in cassandra.cluster.ResponseFuture.result
cassandra.protocol.ConfigurationException: <Error from server: code=2300 [Query invalid because of configuration issue] message="Validation failed:std::_Nested_exception<std::runtime_error> (Could not write key file '/etc/scylla/encrypt_conf/secret_key'): std::filesystem::__cxx11::filesystem_error (error system:13, filesystem error: mkdir failed: Permission denied [/etc/scylla/encrypt_conf/])">
```

File: jenkins-pipelines/longevity-100gb-4h-fips.jenkinsfile (hunk 1 above)
Comment (fruch) [reply jenkins-pipelines/longevity-100gb-4h-fips.jenkinsfile:10]: Why should it work ? 

I'm quite sure we had a few of those in siren-tests

Comment (fgelcer) [reply jenkins-pipelines/longevity-100gb-4h-fips.jenkinsfile:10]: was just an observation...

**Review by fruch — DISMISSED** [review]
LGTM

**Comment by fgelcer** [comment]
@ShlomiBalalis , what about the logevity itself?

File: artifacts_test.py [hunk 2]
```diff
@@ -282,8 +282,34 @@ def check_service_existence(self, service_name):
...
         return True  # exit_status = 1 means the service doesn't exist
 
+    def run_pre_create_schema(self, replication_factor=1):
+        pre_create_schema = self.params.get('pre_create_schema')
+        if pre_create_schema:
+            keyspace_num = self.params.get('keyspace_num')
+            sstable_size = self.params.get('sstable_size')
+            compaction_strategy = self.params.get('compaction_strategy')
+            scylla_encryption_options = self.params.get('scylla_encryption_options')
+            self.log.debug('Pre Creating Schema for c-s with {} keyspaces'.format(keyspace_num))
```
Comment (fgelcer) [inline artifacts_test.py:292]: for logs better use lazy formatting

File: artifacts_test.py [hunk 3]
```diff
@@ -282,8 +282,34 @@ def check_service_existence(self, service_name):
...
+            for i in range(1, keyspace_num+1):
+                keyspace_name = 'keyspace{}'.format(i)
+                self.create_keyspace(keyspace_name=keyspace_name, replication_factor=replication_factor)
+                self.log.debug('{} Created'.format(keyspace_name))
```
Comment (fgelcer) [inline artifacts_test.py:296]: for logs better use lazy formatting

**Review by fgelcer — COMMENTED** [review]
other than nitpick comment, LGTM

File: artifacts_test.py (hunk 2 above)
Comment (ShlomiBalalis) [reply artifacts_test.py:292]: That's what I get for copy-pasing

File: artifacts_test.py (hunk 3 above)
Comment (ShlomiBalalis) [reply artifacts_test.py:296]: That's what I get for copy-pasing

File: artifacts_test.py (hunk 2 above)
Comment (ShlomiBalalis) [reply artifacts_test.py]: Done\

File: artifacts_test.py (hunk 3 above)
Comment (ShlomiBalalis) [reply artifacts_test.py]: Done\

**Comment by ShlomiBalalis** [comment]
Let's wait for the longevity results first

**Comment by ShlomiBalalis** [comment]
So, about the longevity results. Two points:
1. Ended with two error events, but those events are inline with the errors that appear in the latest 4h longevity in enterprise (see a31a600f-e548-433b-9083-bf465fdb0864)
2. It timed out, only by a little. It seems that bootstrap takes longer than the usual run? Which should be in lined with the artifact test, that takes longer than its non-fips counterpart as well (30 minutes opposed to 20)

I'll increase the timeout and rerun, just to see everything will end on the up and up

**Review by fgelcer — APPROVED** [review]
LGTM

**Review by fruch — APPROVED** [review]
LGTM

**Comment by fruch** [comment]
> So, about the longevity results. Two points:
> 
> 1. Ended with two error events, but those events are inline with the errors that appear in the latest 4h longevity in enterprise (see a31a600f-e548-433b-9083-bf465fdb0864)
> 2. It timed out, only by a little. It seems that bootstrap takes longer than the usual run? Which should be in lined with the artifact test, that takes longer than its non-fips counterpart as well (30 minutes opposed to 20)
> 
> I'll increase the timeout and rerun, just to see everything will end on the up and up

@ShlomiBalalis I would recommend opening an issue for the slow startup

**Comment by ShlomiBalalis** [comment]
> > So, about the longevity results. Two points:
> > 
> > 1. Ended with two error events, but those events are inline with the errors that appear in the latest 4h longevity in enterprise (see a31a600f-e548-433b-9083-bf465fdb0864)
> > 2. It timed out, only by a little. It seems that bootstrap takes longer than the usual run? Which should be in lined with the artifact test, that takes longer than its non-fips counterpart as well (30 minutes opposed to 20)
> > 
> > I'll increase the timeout and rerun, just to see everything will end on the up and up
> 
> @ShlomiBalalis I would recommend opening an issue for the slow startup

It could be other factors, like the installations (It's a clean ami), encryption and whatnot. I'll start a job with the same exact setup just with a regular ami (clean ubuntu 20 with no fips) and see if it's the same. It it is, no point in an issue.

The second run passed altogether BTW

File: test-cases/longevity/longevity-encryption-at-rest-200GB-6h.yaml [hunk 4]
```diff
@@ -27,12 +27,3 @@ authenticator_password: cassandra
 authorizer: 'CassandraAuthorizer'
 
 pre_create_schema: True
-scylla_encryption_options: "{ 'cipher_algorithm' : 'AES/ECB/PKCS5Padding', 'secret_key_strength' : 128, 'key_provider': 'LocalFileSystemKeyProviderFactory', 'secret_key_file': '/etc/scylla/encrypt_conf/secret_key'}"
```
Comment (roydahan) [inline test-cases/longevity/longevity-encryption-at-rest-200GB-6h.yaml:30]: Why was this removed?

Comment (fruch) [reply test-cases/longevity/longevity-encryption-at-rest-200GB-6h.yaml:30]: it was moved to the configuration file it belongs, `local-ear.yaml`

**Comment by ShlomiBalalis** [comment]
> > > So, about the longevity results. Two points:
> > > 
> > > 1. Ended with two error events, but those events are inline with the errors that appear in the latest 4h longevity in enterprise (see a31a600f-e548-433b-9083-bf465fdb0864)
> > > 2. It timed out, only by a little. It seems that bootstrap takes longer than the usual run? Which should be in lined with the artifact test, that takes longer than its non-fips counterpart as well (30 minutes opposed to 20)
> > > 
> > > I'll increase the timeout and rerun, just to see everything will end on the up and up
> > 
> > 
> > @ShlomiBalalis I would recommend opening an issue for the slow startup
> 
> It could be other factors, like the installations (It's a clean ami), encryption and whatnot. I'll start a job with the same exact setup just with a regular ami (clean ubuntu 20 with no fips) and see if it's the same. It it is, no point in an issue.



Said run has finished its bootstrap stage, and it seems that yes, the installation/encryption that were added were the major causes of the longer bootstrap.

Bootstrap events of a regular run (no FIPS, scylla ami):
```
2023-07-25 13:48:27.632: (BootstrapEvent Severity.NORMAL) period_type=begin event_id=d4267ec1-9b0b-41d1-ae0c-26634ce0e4e6 node=longevity-100gb-4h-master-db-node-842b880d-2
2023-07-25 13:52:23.608: (BootstrapEvent Severity.NORMAL) period_type=begin event_id=fb741dcd-e0e8-4ddc-9842-8004996b90c7 node=longevity-100gb-4h-master-db-node-842b880d-3
2023-07-25 13:56:13.876: (BootstrapEvent Severity.NORMAL) period_type=begin event_id=e310d42e-749a-436a-b81c-36413a8ced61 node=longevity-100gb-4h-master-db-node-842b880d-4
2023-07-25 14:00:04.739: (BootstrapEvent Severity.NORMAL) period_type=begin event_id=cfdc205b-b60e-4fb1-9dc5-162735fde666 node=longevity-100gb-4h-master-db-node-842b880d-5
2023-07-25 14:03:57.853: (BootstrapEvent Severity.NORMAL) period_type=begin event_id=46efc7b5-4e13-4e33-9631-17545a4ed0bd node=longevity-100gb-4h-master-db-node-842b880d-6
```
~4 minutes each

Bootstrap events of a FIPS + encryption + clean ubuntu 20 ami run:
```
2023-08-07 23:46:08.346: (BootstrapEvent Severity.NORMAL) period_type=begin event_id=1099636a-f36f-4330-90d6-ec5b04b80a80 node=longevity-fips-fips-sho-db-node-15e6157e-2
2023-08-07 23:58:30.199: (BootstrapEvent Severity.NORMAL) period_type=begin event_id=f72be7dc-d4c6-4e5a-b24e-23eb6a68c450 node=longevity-fips-fips-sho-db-node-15e6157e-3
2023-08-08 00:11:43.480: (BootstrapEvent Severity.NORMAL) period_type=begin event_id=0908f626-fa0b-47dc-a030-2dffbb2171b0 node=longevity-fips-fips-sho-db-node-15e6157e-4
2023-08-08 00:24:25.640: (BootstrapEvent Severity.NORMAL) period_type=begin event_id=56fa2f8f-cf85-40b6-b3fe-5aea0ae9a53a node=longevity-fips-fips-sho-db-node-15e6157e-5
2023-08-08 00:36:52.533: (BootstrapEvent Severity.NORMAL) period_type=begin event_id=ca82384d-0310-48af-975b-afab730daae3 node=longevity-fips-fips-sho-db-node-15e6157e-6
```
~12 minutes each

Bootstrap events of encryption + clean ubuntu 20 ami (no FIPS) run:
```
2023-08-08 13:38:27.617: (BootstrapEvent Severity.NORMAL) period_type=begin event_id=7eba8be0-c700-4cef-b0e7-1ba491304642 node=longevity-fips-100gb-4h-db-node-2c5eb893-2
2023-08-08 13:48:18.621: (BootstrapEvent Severity.NORMAL) period_type=begin event_id=beeb2887-e829-43b8-811e-bd6eef56b721 node=longevity-fips-100gb-4h-db-node-2c5eb893-3
2023-08-08 13:58:02.779: (BootstrapEvent Severity.NORMAL) period_type=begin event_id=11cf987c-7b48-4a6e-be28-deb7b4fd4d4d node=longevity-fips-100gb-4h-db-node-2c5eb893-4
2023-08-08 14:07:51.756: (BootstrapEvent Severity.NORMAL) period_type=begin event_id=fb3a48ec-7b32-41c6-82bc-4241d3193ab6 node=longevity-fips-100gb-4h-db-node-2c5eb893-5
2023-08-08 14:17:37.999: (BootstrapEvent Severity.NORMAL) period_type=begin event_id=07dfd6b5-6675-4c9f-96d8-0b8464c9bafe node=longevity-fips-100gb-4h-db-node-2c5eb893-6
```
~10 minutes each

So using the FIPS image does not cause a significant increase in bootstrap (2 minutes for each nodes at most). I don't see an issue with it

@fgelcer @fruch Do you concur?

**Comment by fgelcer** [comment]
> So using the FIPS image does not cause a significant increase in bootstrap (2 minutes for each nodes at most). I don't see an issue with it

yes, this is what we discussed, and if that is the case, then of course, it will take longer than the regular 4h longevity

**Comment by fruch** [comment]
> So using the FIPS image does not cause a significant increase in bootstrap (2 minutes for each nodes at most). I don't see an issue with it
> 
> @fgelcer @fruch Do you concur?

it's still 2min we don't know what's their origin is it part of the installation what is long or part of the scylla boot which is now longer.
anyhow it's not something expected AFAIK, and I think it should documented in an issue (it's not a showstopper)
